float32 vx
float32 vy
float32 wz
float32 epsi
float32 s
float32 ey